### ISSUE TITLE
Automatic build and install when file changes

### ISSUE THREAD

**Issue by ljlin** [ocaml/dune#7112]
Hi all,

I was wondering whether there is a option that can build and install a dune program automatically every time there's a file change?

`dune build --watch --terminal-persistence=clear-on-rebuild` works perfectly; it automatically rebuild the project.

But `dune install --watch --terminal-persistence=clear-on-rebuild` doesn't; watch seems never triggered.

I am using `dune 3.6.2` on `macOS 12.6.3`. I am not sure whether it's a bug or I miss understood the meaning of `dune install -w`.

Thanks for helping.

**Comment by Alizter** [comment]
Watch mode for `dune install` is currently a noop. I suppose we could implement a watch mode for installation, however I am unsure if that is a good idea or not.

**Comment by esope** [comment]
Maybe a warning or an error should be printed when the `--watch` flag is passed to `dune install`?
Or documentation could make the behavior clearer.

**Comment by emillon** [comment]
I agree that the bug is that `--watch` is accepted. It should not. More generally, our CLI arguments are a bit too global and accepted in places where they shouldn't.
